sync-core/src/main/java/com/cloudant/sync/query/OperatorExpressionNode.java

```diff
@@ -0,0 +1,32 @@
+//  Copyright (c) 2014 Cloudant. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+//  except in compliance with the License. You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software distributed under the
+//  License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+//  either express or implied. See the License for the specific language governing permissions
+//  and limitations under the License.
+
+package com.cloudant.sync.query;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ *  An OperatorExpressionNode object is used by the
+ *  {@link com.cloudant.sync.query.UnindexedMatcher}.  Since some methods within the
+ *  {@link com.cloudant.sync.query.UnindexedMatcher} class that utilize an OperatorExpressionNode
+ *  object are static in nature, the OperatorExpressionNode class cannot be implemented as an
+ *  inner class to {@link com.cloudant.sync.query.UnindexedMatcher} and must be
+ *  implemented this way instead.
+ *
+ *  @see com.cloudant.sync.query.UnindexedMatcher
+ */
+public class OperatorExpressionNode implements QueryNode {
```

Comment:
I made the class package protected instead of public.  In order to make it private it would have to be an inner class.
https://github.com/cloudant/sync-android/commit/cf527fd540554a34338857f179c44fa358444ad3#diff-cac498342a06ebddae3efd7af8bbbf5eL28
I did consider moving all of the QueryNode classes to be inner classes of the QueryExecutor.  @mikerhodes what are your thoughts on that?  Would it be better to leave them as top level classes or does it make sense to make them inner to the executor?  I go back and forth with what I think is right.